### ISSUE TITLE
Time condition leaves emtply list when selecting and deselecting a weekday

### ISSUE THREAD

**Issue by TheFes** [home-assistant/frontend#14671]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

If you create a time condition in the UI and then select a weekday, and after that remove it again, it leaves an empty list in the condition:
```yaml
- condition: time
  after: "07:00:00"
  before: "22:00:00"
  weekday: []
```
This causes the condition to fail, as it isn't valid on any weekday now.

### Describe the behavior you expected

I would expect the `weekday:` key to be removed if no day is selected anymore.

### Steps to reproduce the issue

1. Create an automation in the GUI
2. Add a time condtion
3. Select a weekday
4. Remove the selected weekday
...

### What version of Home Assistant Core has the issue?

2022.12.1

### In which browser are you experiencing the issue with?

Google Chrome 108.0.5359.99

### Which operating system are you using to run this browser?

Windows 10

**Comment by TheFes** [comment]
This seems to be a more generic issue, if data in a field is entered, and then removed, the key is left in.
For example when a `value_template` is added on a numeric_state trigger, and the the data is removed in the UI, `value_template: ""` is left in the trigger, causing the trigger not to function.

**Comment by TheFes** [comment]
Also see: https://github.com/home-assistant/frontend/discussions/14838

**Comment by karwosts** [comment]
This is fixed by #14992
